app/src/main/java/com/babylon/wallet/android/domain/usecases/securityproblems/GetSecurityProblemsUseCase.kt

```diff
@@ -0,0 +1,127 @@
+package com.babylon.wallet.android.domain.usecases.securityproblems
+
+import com.babylon.wallet.android.domain.model.SecurityProblem
+import com.babylon.wallet.android.domain.model.toDomainModel
+import com.radixdlt.sargon.AddressesOfEntitiesInBadState
+import com.radixdlt.sargon.BackupResult
+import com.radixdlt.sargon.CheckSecurityProblemsInput
+import com.radixdlt.sargon.extensions.ProfileEntity
+import com.radixdlt.sargon.os.SargonOsManager
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import rdx.works.core.domain.cloudbackup.BackupState
+import rdx.works.core.sargon.factorSourceId
+import rdx.works.core.sargon.isNotHidden
+import rdx.works.profile.domain.backup.GetBackupStateUseCase
+import javax.inject.Inject
+
+class GetSecurityProblemsUseCase @Inject constructor(
+    private val getEntitiesWithSecurityPromptUseCase: GetEntitiesWithSecurityPromptUseCase,
+    private val getBackupStateUseCase: GetBackupStateUseCase,
+    private val sargonOsManager: SargonOsManager
+) {
+
+    operator fun invoke(): Flow<Set<SecurityProblem>> = combine(
+        getEntitiesWithSecurityPromptUseCase(),
+        getBackupStateUseCase()
+    ) { entitiesWithSecurityPrompts, backupState ->
+
+        // active personas that need cloud backup
+        val activePersonasNeedCloudBackup = entitiesWithSecurityPrompts
+            .count { entityWithSecurityPrompt ->
+                entityWithSecurityPrompt.entity.isNotHidden() &&
+                    entityWithSecurityPrompt.entity is ProfileEntity.PersonaEntity &&
+                    (
+                        entityWithSecurityPrompt.prompts.contains(SecurityPromptType.CONFIGURATION_BACKUP_PROBLEM) ||
+                            entityWithSecurityPrompt.prompts.contains(SecurityPromptType.WALLET_NOT_RECOVERABLE) ||
+                            entityWithSecurityPrompt.prompts.contains(SecurityPromptType.CONFIGURATION_BACKUP_NOT_UPDATED)
+                        )
+            }
+
+        sargonOsManager.sargonOs.checkSecurityProblems(
```

Comment:
Both `sargonOs` getter and `checkSecurityProblems` method on it are throwing. Please ensure the exception is properly handled.

Reply:
yes good catch! thank you

Reply:
@sergiupuhalschi-rdx what do you recommend about handling the exception?
adding a `runCatching` in the usecase and return `Flow<Result<Set<SecurityProblem>>>` 
or adding a `.catch { Timber.e("failed to get security problems...") }` in the viewmodel where I collect the flows?

I’m leaning toward the second option, tbh.

Reply:
If we should just propagate the error with no additional logic then `catch` looks like a better option. You might also consider adding `catch` inside the `useCase` instead of each `viewModel` using this `useCase`. Either option sounds good.